tasks/conf.yml

```diff
@@ -64,21 +64,18 @@
   tags:
     - config-additional
 
-- name: Upstart check
-  stat: path=/etc/init/
-  register: etc_init_check
-
-- name: systemd check
-  stat: path=/usr/lib/systemd/system/
-  register: systemd_check
-
 - name: Upstart environment variables
   lineinfile: dest=/etc/init/marathon.conf backup=yes state=present create=true insertbefore='exec.*' line="env {{ item }}"
   with_items: "{{ marathon_env_vars }}"
-  when: etc_init_check.stat.exists == true
+  when:
+    - ansible_service_mgr != 'systemd'
   notify: Restart marathon
 
+- name: Create /etc/sysconfig directory
+  file: path=/etc/sysconfig state=directory owner=root group=root mode=0755
```

Comment:
You are right I doubt they are used on Ubuntu maybe even on RedHat as well. In my current company I was recently migrating init to systemd and was setuping service.d folders as well so this might be good option to implement it. Systemd supports EnvironmentFile as well but since we use prepacked service it might not be an option here.